src/tests/unittests/asn1.c

```diff
@@ -47,12 +47,91 @@ static void torture_large_oid(void **state)
 	assert_int_equal(rv, SC_ERROR_NOT_SUPPORTED);
 }
 
+static void torture_integer(void **state)
+{
+	/* Without the Tag and Length {0x02, 0x01} */
+	u8 zero[] = {0x00};
+	u8 one[] = {0x01};
+	u8 minus_one[] = {0xFF};
+	u8 max2[] = {0x7F, 0xFF};
+	u8 min2[] = {0x80, 0x00};
+	u8 max4[] = {0x7F, 0xFF, 0xFF, 0xFF};
+	u8 min4[] = {0x80, 0x00, 0x00, 0x00};
+	u8 over[] = {0x7F, 0xFF, 0xFF, 0xFF, 0xFF};
```

Comment:
negative test cases missing:
zero encoded with no bytes: `{};`
zero encoded with multiple bytes: `0x00, 0x00`
one encoded with multiple bytes: `0x00, 0x01`
minus one encoded with multiple bytes: `0xff, 0xff`

Reply:
but then we should also have a test case to verify that values padded so that they don't look negative, are accepted, as in: `0x00, 0x80` is decoded to 128 but `0x00, 0x7f` causes an error